Server/Communication/router.py

```diff
@@ -111,18 +111,30 @@ def processRequest(message):
 		})
 		return
 
+	# Store the Request ID
+	request_id = None
+	if 'Request_ID' in data:
+		request_id = data['Request_ID']
```

Comment:
Can we initialize resp after this line and then insert the request id if it exists?  That way the following code won't need to be duplicated many times:
if request_id:
 +			resp['Request_ID'] = request_id

Could use append instead of initializing the resp dictionary in each further instance.